[RNMobile] Page Template Picker UI Improvements

geriux opened the pull request:
`Gutenberg-mobile` PR -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/1720

## Description
This PR updates the current styles for the Page Template picker. It also includes some code changes to support more items in the list and accessibility.

## To test
- Open WP iOS / WP Android
- Go to **Site Pages**
- Add a new page
  - For iOS Tap on the Add button in the header (top right)
- **Expect** the page template picker to be shown with the styles from the screenshots
- Tap on either predefined layout
- **Expect** the content of the page to change and the picker to **disappear**

| iOS | Android |
|     :---:      |     :---:      |
| <img src="https://user-images.githubusercontent.com/4885740/71714968-13841880-2e10-11ea-9339-2c9de1b4c513.gif" width=250 /> | <img src="https://user-images.githubusercontent.com/4885740/71714988-27c81580-2e10-11ea-8ddc-d5d0db6855ad.gif" width=270 />  |

## Screenshots

### iOS 

| Default | Dark mode |
|     :---:      |     :---:      |
| <img src="https://user-images.githubusercontent.com/4885740/71715020-40383000-2e10-11ea-98a1-354d7c6e752f.png" width=250 /> | <img src="https://user-images.githubusercontent.com/4885740/71715023-4201f380-2e10-11ea-8199-bb864888657f.png" width=250 />  |

| iPad Default | iPad Dark mode |
|     :---:      |     :---:      |
| <img src="https://user-images.githubusercontent.com/4885740/71715039-580fb400-2e10-11ea-8b82-ef4d59540369.png" width=250 /> | <img src="https://user-images.githubusercontent.com/4885740/71715041-59d97780-2e10-11ea-84d0-270a399b338b.png" width=250 />  |

### Android

<img src="https://user-images.githubusercontent.com/4885740/71715070-6cec4780-2e10-11ea-9a10-402520419053.png" width=250 /> 

## Types of changes
- UI Styling
- Accessibility

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards.
- [x] My code has proper inline documentation.
- [x] I've included developer documentation if appropriate.
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

Commit:
RNMobile - Page template picker: styles and icons, scroll and accessibility text

geriux commented on packages/block-editor/src/components/page-template-picker/button.native.js:
```diff
@@ -1,20 +1,47 @@
+	const butonTextStyles = getStylesFromColorScheme( styles.buttonText, styles.buttonTextDark );
+	const accessibilityLabel = sprintf(
+		/* translators: accessibility text. Inform about list of predefined layout options. %1$s: Layout name, e.g About. */
+		__( 'Predefined layout picker. %1$s' ),
```
After reading this https://make.wordpress.org/design/2019/11/14/blocks-patterns-and-layouts/ I used `Layout` instead of `Page template`.

koke replied:
Nice, I wonder if we should start renaming things more broadly (file and component names)

koke replied:
This seems like too much text before the actual button label. I'm not sure I would specify an `accessibilityLabel` here. Instead, we can add an `accessibilityHint="Double tap to select layout"`.

Ideally we should find a way to convey that this is a list of possible layouts rather than individual buttons, but I don't see any support for [accessibility containers](https://developer.apple.com/documentation/uikit/accessibility/uiaccessibilitycontainer) in React Native yet. Since we only have a couple buttons now, it's not a big deal, but something to keep in mind for the future.

geriux replied:
Good idea! I'll use `accessibilityHint` instead. I hope they add support for that, it'd be super great.

geriux replied:
Updated https://github.com/WordPress/gutenberg/pull/19307/commits/4c1c32684840fe0585d7203cd119c08fa13f6db9

Commit:
Update Page templates picker for web

koke commented:
@iamthomasbishop are we going with [Twemoji](https://github.com/twitter/twemoji) for the icons or should we use the native emoji on each platform?

If we are, we should probably add `twemoji` as a proper dependency and try to get the SVG from there, or at least ensure we do [proper attribution](https://github.com/twitter/twemoji#license).

koke commented on packages/edit-post/src/components/layout/index.native.js:
```diff
@@ -129,10 +129,10 @@ class Layout extends Component {
 						parentHeight={ this.state.rootViewHeight }
 						style={ toolbarKeyboardAvoidingViewStyle }
 					>
+						{ showPageTemplatePicker && <__experimentalPageTemplatePicker /> }
```
In #19106 I was also moving the picker to the bottom, although I guess it makes more sense to do it as part of this PR.

koke reviewed (approved):
This is looking pretty nice 👌 From my side, it's good to go after the accessibility changes. I can't request a review from @iamthomasbishop directly, but I'd wait for him to check that the changes look right

geriux commented:
> This is looking pretty nice 👌 From my side, it's good to go after the accessibility changes. I can't request a review from @iamthomasbishop directly, but I'd wait for him to check that the changes look right

Thanks!! Yup, I'll wait to check with @iamthomasbishop about your [comment](https://github.com/WordPress/gutenberg/pull/19307#issuecomment-569035136) above about the icons and that the changes look ok. 👍

Commit:
RNMobile - Layout template picker: accessibility hint added

iamthomasbishop commented:
Hey @geriux – looking good! I just have a few notes:

* I was going to say the chip/button background seemed a bit too dark, then I realized I accidentally bumped the background colors to a stronger shade of gray 🤦‍♂ I've updated the components in the design docs. (black @ 4% in light mode, white @ 8% in dark mode)

* Ideally, we can just use the native emoji if possible // cc @koke 

* What happens to the chips/buttons if I hide the keyboard? Do they stay fixed in the middle of the screen, or move down with the keyboard/toolbar?

Commit:
RNMobile - Layout template picker: style improvements, removed unneeded icons

geriux commented:
> Hey @geriux – looking good! I just have a few notes:

Thanks for reviewing it!

> * I was going to say the chip/button background seemed a bit too dark, then I realized I accidentally bumped the background colors to a stronger shade of gray 🤦‍♂ I've updated the components in the design docs. (black @ 4% in light mode, white @ 8% in dark mode)

Haha :D no problem, I've just updated the code and the screenshots above, let me know if they look ok now.

> * Ideally, we can just use the native emoji if possible // cc @koke

Changed for the native emoji

> * What happens to the chips/buttons if I hide the keyboard? Do they stay fixed in the middle of the screen, or move down with the keyboard/toolbar?

Right now they move down, let me know if this is ok or if they should stay fixed:

<img align="center" src="https://user-images.githubusercontent.com/4885740/71715353-5c889c80-2e11-11ea-9985-61aabf3e514e.gif" width=250 />

Commit:
Page layout picker: remove Icon dependency

iamthomasbishop commented:
@geriux awesome! Colors, emoji, and positioning are  all looking great!
